lib/data/network/media/media_api.dart

```diff
@@ -34,4 +35,23 @@ class UploadFileAPI {
 
     return UploadFileResponse.fromJson(response);
   }
+
+  Future<UrlPreviewResponse> getUrlPreview({
+    required Uri uri,
+    int? preferredPreviewTime,
+  }) async {
+    final response = await _client
+        .get(
+          HomeserverEndpoint.getPreviewUrlServicePath
+              .generateHomeserverIdentityEndpoint(),
+          queryParameters: {
```

Comment:
the API maybe rate limit, what you do in that case

Reply:
Throw exception. Currently same logic is `upload` method :)).

I'll create a ticket handle error handling from server

Reply:
I think we should cache 429 for preventing client try to request too much to server